src/wp-includes/class-wp-theme-json.php

```diff
@@ -766,14 +776,46 @@ private function get_block_classes( $style_nodes ) {
 			$settings     = _wp_array_get( $this->theme_json, array( 'settings' ) );
 			$declarations = self::compute_style_properties( $node, $settings );
 
-			// 1. Separate the ones who use the general selector
-			// and the ones who use the duotone selector.
-			$declarations_duotone = array();
+			// Filter the style $declarations.
+			$declarations_duotone   = array();
+			$declarations_block_gap = array();
 			foreach ( $declarations as $index => $declaration ) {
+				// 1. Separate the ones who use the general selector
+				// and the ones who use the duotone selector.
 				if ( 'filter' === $declaration['name'] ) {
 					unset( $declarations[ $index ] );
 					$declarations_duotone[] = $declaration;
 				}
+				// 2. Save the block gap row and column CSS vars
+				if ( str_starts_with( $declaration['name'], '--wp--style--block' ) ) {
+					$declarations_block_gap[ $declaration['name'] ] = $declaration;
+					unset( $declarations[ $index ] );
+				}
+			}
+
+			if ( ! empty( $declarations_block_gap ) ) {
```

Comment:
This is the first attempt, which will be the worst attempt. 

Just getting things to work as expected before refactoring.